### PR TITLE
added: complex function to paddle frontend

### LINKED ISSUE #21085: complex

### PR DESCRIPTION
**Pull request by ahmedlila**
Close #21085

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 1]
```diff
@@ -178,3 +178,23 @@ def diag(x, offset=0, padding_value=0, name=None):
...
     return ivy.logspace(start, stop, num=num, base=base, dtype=dtype)
+
+
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float32", "float64")},
+    "paddle",
+)
+@to_ivy_arrays_and_back
+def complex(real, imag, name=None):
+    assert ivy.dtype(real) == ivy.dtype(imag), (
```
Comment (MahmoudAshraf97) [inline ivy/functional/frontends/paddle/tensor/creation.py:189]:
```suggestion
    assert real.dtype == imag.dtype, (

```

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 2]
```diff
@@ -178,3 +178,23 @@ def diag(x, offset=0, padding_value=0, name=None):
...
+        "(InvalidArgument) The type of data we are trying to retrieve does not match"
+        " the type of data currently contained in the container."
+    )
+    complex_dtype = (
+        ivy.ComplexDtype("complex64")
+        if ivy.dtype(real) == "float32"
+        else ivy.ComplexDtype("complex128")
```
Comment (MahmoudAshraf97) [inline ivy/functional/frontends/paddle/tensor/creation.py:196]:
```suggestion
        "complex64"
        if real.dtype == "float32"
        else "complex128"

```

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

**Review by MahmoudAshraf97 — APPROVED** [review]

**Comment by MahmoudAshraf97** [comment]
ivy-gardener

**Pull request merged**